src/appleseed/renderer/kernel/rendering/generic/genericsamplerenderer.cpp

```diff
@@ -253,6 +258,17 @@ namespace
                 if (!shading_point_ptr->hit_surface())
                     break;
 
+                // Check if matte shader is enabled and apply alpha and color if it is.
+                const Material* material = shading_point_ptr->get_material();
+                if (material != nullptr &&
+                    material->get_render_data().m_shader_group != nullptr &&
+                    material->get_render_data().m_shader_group->has_matte())
```

Comment:
This test is not correct. has_matte() means that the result of executing the shadergroup may contain matte closures, 
not that it always contains matte closures. We need to get that information from the shading engine after executing the shader and knowing for sure that as_matte() is present.